### ISSUE TITLE
How can I extract the gradients w.r.t. the output layer?

### ISSUE THREAD

**Issue by tserdar** [ultralytics/yolov5#3679]
## ❔Context
I am trying to implement [BADGE](https://openreview.net/pdf?id=ryghZJBKPS) with YOLOv5. How can I extract the gradients w.r.t to the final layer? 
![image](https://user-images.githubusercontent.com/62431982/122539103-83e42100-d027-11eb-9008-1d359113e9d7.png)
(Taken from [paper](https://openreview.net/pdf?id=ryghZJBKPS) )

## Question:
1. Assume that there is an already trained .pt file. And there is a single image with annotation to train (validation set is the original one). 
2. How can I extract the gradient embeddings given above for this single image in training mode?  

`Apparently now the loss function is different.`
